connector/web3/web3.go

```
@@ -68,6 +83,42 @@ func (c *web3Connector) Verify(address, msg, signedMsg string) (identity connect
 	return identity, nil
 }
 
+func (c *web3Connector) VerifyERC1271Signature(contractAddress common.Address, hash, signature []byte) (identity connector.Identity, err error) {
+	if c.ethClient == nil {
+		c.logger.Errorf("Eth client was not initialized successfully %v", err)
+		return identity, errors.New("error occurred completing authentication, please try again")
+	}
+	var msgHash [32]byte
+	copy(msgHash[:], hash)
+
+	// ContractLogin is just a simple interface with the signature for IsValidSignature
+	/**
+	 * function isValidSignature(bytes32 hash, bytes memory signature) external view returns (bytes4 magicValue);
+	 */
+	ct, err := NewContractLogin(contractAddress, c.ethClient)
+	if err != nil {
+		return identity, errors.New("error occurred completing login")
```

I would bubble up the error.